contracts/diaspore/DebtEngine.sol

```diff
@@ -205,7 +220,43 @@ contract DebtEngine is ERC721Base {
             _tokens: paidToken
         });
     }
-    
+
+    function payBatch(
+        bytes32[] _ids,
+        uint256[] _amounts,
+        address _oracle,
+        bytes8 _currency,
+        bytes _oracleData
+    ) external returns (uint256[], uint256[]) {
+        uint256 count = _ids.length;
+        require(count == _amounts.length, "The loans and the amounts do not correspond.");
+        require(count > 0, "There are not loans to pay.");
+
+        uint256[] memory data = new uint256[](3);
```

Comment:
You can use uint256[3] data to spend less gas and the _pay function take a uint256[3] data